### PR TITLE
feat: Logging Integration + Code Cleanup

### PR DESCRIPTION
**Pull request by vineetshar**
- [x] Tested on local & docker
- [-] Documentation Updated
- [x] Attached required context / screenshot

This PR helps replace print() with logging , so that it supports integration with any observability / logging platform easily.

### CODE REVIEW

File: server/endpoint_detection.py [hunk 1]
```diff
@@ -163,7 +164,7 @@ def identify_django_endpoints(self, project_path, project_id):
                                             view_type,
                                         )
                                     )
-                                    print(file_path, identifier)
+                                    logging.info("project_id:" , project_id ,"identify_django_endpoints -> file_path : ",file_path, " and " , "identifier : ",identifier)
```
Comment (dhirenmathur) [inline server/endpoint_detection.py:167]: Can we use normal fstrimgs everywhere please

File: server/endpoint_detection.py [hunk 2]
```diff
@@ -153,7 +154,7 @@ def identify_django_endpoints(self, project_path, project_id):
                                         if not view_name.endswith("as_view")
                                         else view_name.rsplit(".", 1)[0]
                                     )
-                                    print(view_name)
+                                    logging.info("project_id:" , project_id ,"identify_django_endpoints -> view_name : ",view_name )
```
Comment (dhirenmathur) [inline server/endpoint_detection.py:157]: Can we use normal fstrimgs everywhere please

File: server/endpoint_detection.py [hunk 3]
```diff
@@ -531,9 +532,9 @@ def delete_endpoints(self, project_id, user_id):
                 "DELETE FROM endpoints WHERE project_id = %s AND project_id IN (SELECT id FROM projects WHERE user_id = %s)",
                 (project_id, user_id))
             conn.commit()
-            print(f"Endpoints with project_id {project_id} and user_id {user_id} deleted successfully.")
+            logging.info(f"project_id: ,{project_id}, Endpoints with project_id {project_id} and user_id {user_id} deleted successfully.")
         except psycopg2.Error as e:
-            print("An error occurred while deleting endpoints:", e)
+            logging.error("project_id:" , project_id ,"An error occurred while deleting endpoints:", e)
```
Comment (dhirenmathur) [inline server/endpoint_detection.py:537]: Fstrimg

File: server/plan.py [hunk 4]
```diff
@@ -108,8 +109,8 @@ async def _plan(
 To help integration test the flow above:
 1. Analyze the provided code and explaination.
 2. List diverse happy path and edge case scenarios that the function should handle. 
-3. Include exactly 3 scenario statements of happpy paths and 3 scenarios of edge cases. 
-4. Format your output in JSON format as such, each scenario is only a string statement:
+3. Include a MINIMUM of 3 and MAXIMUM of 6 happy path scenarios and 3 edge case scenarios.
```
Comment (dhirenmathur) [inline server/plan.py:112]: Regression on prompt, please update the string clause atleast

File: server/plan.py [hunk 5]
```diff
@@ -219,7 +220,7 @@ async def run_tests(self, identifier, content):
             output = result.stdout if result.stdout else result.stderr
             return output
         except Exception as e:
-            print(e)
+            logging.error("error in run_tests ",e)
```
Comment (dhirenmathur) [inline server/plan.py:223]: Why do we have run test in core

File: server/router.py [hunk 6]
```diff
@@ -162,7 +163,8 @@ def get_blast_radius_details(
             additions = sum(file.additions for file in git_diff.files)
             deletions = sum(file.deletions for file in git_diff.files)
             lines_impacted = additions + deletions
-            print({
+            logging.info({
```
Comment (dhirenmathur) [inline server/router.py:166]: Fstring?

File: server/utils/ai_helper.py [hunk 7]
```diff
@@ -41,7 +42,8 @@ def print_messages(
         role = message.type
         color_prefix = color_prefix_by_role[role]
         content = message.content
-        print(f"{color_prefix}\n[{role}]\n{content}")
+        logging.info("print_messages")
```
Comment (dhirenmathur) [inline server/utils/ai_helper.py:45]: Can be deleted

File: server/utils/ai_helper.py [hunk 8]
```diff
@@ -50,9 +52,10 @@ def print_message_delta(
     """Prints a chunk of messages streamed back from GPT."""
     role = delta.type
     color_prefix = color_prefix_by_role[role]
-    print(f"{color_prefix}\n[{role}]\n", end="")
+    logging.info("print_message_delta")
```
Comment (dhirenmathur) [inline server/utils/ai_helper.py:55]: Can be deleted

**Review by dhirenmathur — DISMISSED** [review]

File: server/endpoint_detection.py (hunk 1 above)
Comment (vineetshar) [reply server/endpoint_detection.py]: done

File: server/utils/ai_helper.py (hunk 8 above)
Comment (vineetshar) [reply server/utils/ai_helper.py]: done

File: server/utils/ai_helper.py (hunk 7 above)
Comment (vineetshar) [reply server/utils/ai_helper.py]: done

File: server/router.py (hunk 6 above)
Comment (vineetshar) [reply server/router.py]: done

File: server/plan.py (hunk 5 above)
Comment (vineetshar) [reply server/plan.py]: removed

File: server/plan.py (hunk 4 above)
Comment (vineetshar) [reply server/plan.py]: fixed

**Review by cash1998 — APPROVED** [review]
